### PR TITLE
Fix ternary reference input metadata

### LINKED ISSUE #189544: [OpInfo] Ternary reference inputs missing important metadata
### 🐛 Describe the bug

`reference_inputs_elementwise_ternary` generates reference `SampleInput`s for `addcdiv`, `addcmul`, and `clamp`. These generated samples have two problems:
1. Generated samples have their `SampleInput.broadcasts_input` at its default value of `False`, even when it should not.
2. For `clamp`, broadcasted tensor bounds are generated with `requires_grad=True`, but backward fails when the gradient must be reduced to the original bound shape.

This is a test metadata issue, not an operator correctness issue. This metadata is used to avoid running in-place variants when broadcasting would require resizing the first input. Because the metadata is incorrect, the vmap exhaustive tests fail with a broadcast-shape error, which is why they have been added as [xfails](https://github.com/pytorch/pytorch/blob/main/test/functorch/test_vmap.py#L4488).

### Versions

PyTorch version: 2.14.0a0+gitb9655f2
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.3 LTS (aarch64)
GCC version: (Ubuntu 13.3.0-6ubuntu2~24.04.1) 13.3.0
Clang version: Could not collect
CMake version: version 3.31.6
Libc version: glibc-2.39

Python version: 3.12.3 (main, Jun 19 2026, 12:46:00) [GCC 13.3.0] (64-bit runtime)
Python platform: Linux-6.8.0-124-generic-aarch64-with-glibc2.39
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: False
Caching allocator config: N/A

CPU:
Architecture:                            aarch64
CPU op-mode(s):                          32-bit, 64-bit
Byte Order:                              Little Endian
CPU(s):                                  19
On-line CPU(s) list:                     0-18
Vendor ID:                               ARM
Model name:                              Neoverse-N1
Model:                                   1
Thread(s) per core:                      1
Core(s) per socket:                      19
Socket(s):                               1
Stepping:                                r3p1
BogoMIPS:                                50.00
Flags:                                   fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm lrcpc dcpop asimddp ssbs
NUMA node(s):                            1
NUMA node0 CPU(s):                       0-18
Vulnerability Gather data sampling:      Not affected
Vulnerability Indirect target selection: Not affected
Vulnerability Itlb multihit:             Not affected
Vulnerability L1tf:                      Not affected
Vulnerability Mds:                       Not affected
Vulnerability Meltdown:                  Not affected
Vulnerability Mmio stale data:           Not affected
Vulnerability Reg file data sampling:    Not affected
Vulnerability Retbleed:                  Not affected
Vulnerability Spec rstack overflow:      Not affected
Vulnerability Spec store bypass:         Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:                Mitigation; __user pointer sanitization
Vulnerability Spectre v2:                Mitigation; CSV2, BHB
Vulnerability Srbds:                     Not affected
Vulnerability Tsa:                       Not affected
Vulnerability Tsx async abort:           Not affected
Vulnerability Vmscape:                   Not affected

Versions of relevant libraries:
[pip3] mypy==1.16.0
[pip3] mypy_extensions==1.1.0
[pip3] numpy==1.26.4
[pip3] nvidia-cuda-cupti==13.3.75
[pip3] onnx==1.21.0
[pip3] onnx-ir==0.1.16
[pip3] onnxscript==0.6.2
[pip3] optree==0.13.0
[pip3] torch==2.14.0a0+gitb9655f2
[conda] Could not collect

cc @mruberry @zou3519 @Chillee @samdow @kshitij12345

### PR DESCRIPTION
**Pull request by GodlyDonuts**
Fixes #189544.

Marks ternary reference samples whose first input is broadcast, so vmap skips invalid in-place variants. For clamp, broadcasted tensor bounds no longer request gradients that cannot be reduced to their original shapes.

Removes the resolved vmap xfails for addcdiv, addcmul, and clamp.

Validation:
- `python3 -m py_compile torch/testing/_internal/common_methods_invocations.py test/functorch/test_vmap.py`
- Targeted operator smoke check covering all generated ternary reference cases and their valid in-place variants`

### CODE REVIEW

**Comment by zou3519** [comment]
@claude review this please and put the review inside a GitHub collapsible section

**Review by zou3519 — CHANGES_REQUESTED** [review]
we shouldn't be changing the opinfo inputs

File: test/functorch/common_utils.py [hunk 1]
```diff
@@ -135,11 +135,13 @@ def is_valid_inplace_sample_input(sample_input, op, inplace_variant):
     if not isinstance(sample_input.input, torch.Tensor):
         return False
 
-    # Check if input's dtype matches the output's dtype
+    # In-place variants cannot change the input's shape or dtype.
```
Comment (zou3519) [inline test/functorch/common_utils.py]: why didn't broadcasts_input catch this?

**Comment by GodlyDonuts** [comment]
broadcasts_input did not catch these cases because reference_inputs_elementwise_ternary never populated it, so it defaulted to False. I moved the fix to the metadata source: the regular, noncontiguous, and type-promotion samples now set broadcasts_input without changing the generated tensors. I also reverted the output-shape fallback in is_valid_inplace_sample_input. Fixed in 613352ed62.

**Comment by GodlyDonuts** [comment]
@zou3519 Please take a look when you are free.
